### ISSUE TITLE
NoSuchMethodError: method not found

### ISSUE THREAD

**Issue by dariowskii** [shamblett/mqtt5_client#111]
In our web build we encountered this error many many times.

<img width="842" alt="Screenshot 2024-09-11 alle 13 04 37" src="https://github.com/user-attachments/assets/4205c48a-8f30-441d-aee8-de0a636e877f">

the `client` variable is dynamic and I think in web the `destroy` method is not implemented.

**Comment by shamblett** [comment]
Yes you are correct, the destroy method is used only on the Socket class. The mqtt_client package does not call destroy during disconnect in the web build. I'll update this package accordingly.

**Comment by shamblett** [comment]
OK, the real fix for this from the mqtt_client package is a rather involved refactoring of the connection handling classes to use generics rather than rely on dynamic typing. This is too large of a change for me to implement and test at the moment.

However, I understand that this is somewhat of a pain so for now I've patched the package to trap the OnNoSuchMethod exception if it is raised and call client.close().

Package re released at version 4.5.3, please try this out.
